src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,901 @@
+            FileSystem.CreateDirectory(FullPathToNewDirectory)
+            Assert.True(IO.Directory.Exists(FullPathToNewDirectory))
+            Dim testFileSource As String = CreateTestFile(TestBase, SourceData, IO.Path.Combine(FullPathToNewDirectory, "TestFile"))
+
+            Assert.True(IO.File.Exists(testFileSource))
+            Assert.Throws(Of IO.IOException)(Sub() FileSystem.DeleteDirectory(FullPathToNewDirectory, DeleteDirectoryOption.ThrowIfDirectoryNonEmpty))
+            Assert.True(IO.Directory.Exists(FullPathToNewDirectory))
+            Assert.True(IO.File.Exists(testFileSource))
+            TestBase.Dispose()
+        End Sub
+
+        <Fact(Skip:="Not Implemented")>
+        Public Sub DeleteDirectory_Directory_UIOption_RecycleOption()
+
+        End Sub
+
+        <Fact(Skip:="Not Implemented")>
+        Public Sub DeleteDirectory_Directory_UIOption_RecycleOption_UICancelOption()
+
+        End Sub
+
+        <Fact>
+        Public Sub DeleteFile_File()
+            Dim TestBase As New FileIOTestBase
+            Dim testFileSource As String = CreateTestFile(TestBase, SourceData)
+
+            Assert.True(IO.File.Exists(testFileSource))
+            FileSystem.DeleteFile(testFileSource)
+            Assert.False(IO.File.Exists(testFileSource))
+            TestBase.Dispose()
+        End Sub
+
+        <Fact(Skip:="Not Implemented")>
+        Public Sub DeleteFile_File_UIOption_RecycleOption()
+
+        End Sub
+
+        <Fact(Skip:="Not Implemented")>
+        Public Sub DeleteFile_File_UIOption_RecycleOption_UICancelOption()
+
+        End Sub
+
+        <Fact>
+        Public Sub DirectoryExists_directory()
+            Dim TestBase As New FileIOTestBase
+            Dim TestDirectory As String = TestBase.TestDirectory()
+            Assert.True(FileSystem.DirectoryExists(TestDirectory))
+            Assert.False(FileSystem.DirectoryExists(IO.Path.Combine(TestDirectory, "NewDirectory")))
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub FileExists_File()
+            Dim TestBase As New FileIOTestBase
+            Dim testFileSource As String = CreateTestFile(TestBase, SourceData)
+            Assert.True(FileSystem.FileExists(testFileSource))
+            FileSystem.FileExists(testFileSource)
+            IO.File.Delete(testFileSource)
+            Assert.False(FileSystem.FileExists(testFileSource))
+            TestBase.Dispose()
+        End Sub
+
+        <Fact>
+        Public Sub FindInFiles_Directory_ContainsText_IgnoreCase_SearchOption_FileWildcards()
+
+        End Sub
+
+        <Fact>
+        Public Sub FindInFiles_Directory_ContainsText_IgnoreCase_SearchType_SearchOption()
+
+        End Sub
+
+        <Fact>
+        Public Sub GetDirectories_Directory()
+            Dim TestBase As New FileIOTestBase
+            Dim DirectoryList As ReadOnlyCollection(Of String) = FileSystem.GetDirectories(TestBase.TestDirectory)
+            Assert.True(DirectoryList.Count = 0)
```

Comment:
Dont directly do tests on TestDirectory. some other tests may interfere with it. Just a create a newdirectory in this and do tests on that

Reply:
TestDirectory is not what you think. Every  TestBase has a unique TestDirectory based on TestName and other factors. I didn't create FileIOTestBase it is from CoreFx IO Tests.